### ISSUE TITLE
"xCAT Object Name Format" error when trying to name osimages starting with "rhels7.3"

### ISSUE THREAD

**Issue by mew2057** [xcat2/xcat-core#2157]
When I create an image with the mkdef -z directive and the following name "rhels7.3-diskless-test" I receive the following error message:

> Error: The object name 'rhels7.3-diskless-test' is invalid, please refer to "man xcatdb" for the valid "xCAT Object Name Format"

The following other image names were considered valid using the same process:

> diskless  (osimage)
diskless-test  (osimage)
diskless91012  (osimage)
diskless9101214161820  (osimage)

It appears that the if the rhels7.3 token is present this error will persist.

**Comment by pdlun92** [comment]
Experimenting further with `mkdef -z` yields following results:
`rhels7.2-GA-ppc64le-RHEV4-install-compute` `rhels7.3-test` and `rhels7.3` return with the error

`rhels-GA-ppc64le-RHEV4-install-compute` `rhels72-GA-ppc64le-RHEV4-install-compute` `rhels7-GA-ppc64le-RHEV4-install-compute` and `rhels-test` are accepted

mkdef seems to be only command affected, as chdef from a name accepted by `mkdef -z` to one that isn't will successfully change the name.  Example:
`rhels7.3-test` is not accepted by mkdef -z
`rhels-test` IS accepted by mkdef -z
then
```[root@fs4 osimage]# chdef -t osimage rhels-test -n rhels7.3-test
Changed the object name from rhels-test to rhels7.3-test.
[root@fs4 osimage]# lsdef -t osimage -o rhels7.3-test
Object name: rhels7.3-test
```

furthermore, running `mkdef` with `-u` option will run successfully with `rhels7.3-test` as name while `mkdef -z` will NOT
```[root@fs4 ~]# mkdef rhels7.3-test -u profile=compute provmethod=statelite
1 object definitions have been created or modified.
```

In short, it appears that the stanza `-z` option for mkdef osimage is unable to correctly parse `.`

**Comment by pdlun92** [comment]
Traced this back to #2134 

Added print statements to indicate what `$objtype` was being passed to/evaluated in `sub isobjnamevalid` 

```
[root@fs4 osimage]# export XCATBYPASS=TRUE
[root@fs4 osimage]# cat test.stanza | mkdef -z
objtype =  BEFORE
objtype = node AFTER
Error: The object name 'rhels7.3-test' is invalid, please refer to "man xcatdb" for the valid "xCAT Object Name Format"
```

This indicates that mkdef -z with a stanza file of object type of osimage is passing a null objtype which is being changed to `node` by the line
```
$objtype="node" unless(defined $objtype and ($objtype ne ""));
```
At which point it is marked as an invalid node name by this line
```
if(($objname !~ /^[a-zA-Z0-9-_]+$/) and !xCAT::NetworkUtils->isIpaddr($objname)){
            return 0;
```
when it fact it is a valid osimage name.  

somehow, mkdef -z needs to recognize and pass the correct objtype so that osimage names are not evaluated as invalid

For completeness, here is the contents of `test.stanza` referenced above
```
# <xCAT data object stanza file>

rhels7.3-test:
    objtype=osimage
    imagetype=linux
    osarch=ppc64le
    osdistroname=rhels7.3-GA-ppc64le
    osname=Linux
    osvers=rhels7.3-GA
    otherpkgdir=/install/post/otherpkgs/rhels7.3-GA/ppc64le/RHEV4/4.0-GA
    otherpkglist=/install/custom/rhels7.3-GA/ppc64le/rhev4.otherpkgs.pkglist
    pkgdir=/install/rhels7.3-GA/ppc64le
    pkglist=/install/custom/rhels7.3-GA/ppc64le/rhev4.pkglist
    profile=compute
    provmethod=install
    template=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl

```

**Comment by whowutwut** [comment]
@tingtli  do we know why this was not caught in the FVT automation?  We have numerous references to `lsdef -t osimage -o <osimage> | sed 's/install-compute:/somename-compute:/g' | mkdef -z` in our documentation.  So this was supported in the past... but with the recent change it is broken...

**Comment by immarvin** [comment]
fixed in fixed in https://github.com/xcat2/xcat-core/issues/2157

**Comment by cxhong** [comment]
maybe chdef also relate to this, I couldn't remove one of postscripts from node.
`````
# lsdef -t node -o frame23cn17 -i postscripts
Object name: frame23cn17…/install/MOFED/MLNX_OFED_LINUX-3.4-1.0.0.4-rhel7.3-ppc64le.iso,mlnxofed_ib_install -p /install/mofed/MLNX_OFED_LINUX-3.4-1.0.0.4-rhel7.3-ppc64le.iso

# chdef -t node -o frame23cn17 -p postscripts=
1 object definitions have been created or modified.

# lsdef -t node -o frame23cn17 -i postscripts
Object name: frame23cn17…/install/MOFED/MLNX_OFED_LINUX-3.4-1.0.0.4-rhel7.3-ppc64le.iso,mlnxofed_ib_install -p /install/mofed/MLNX_OFED_LINUX-3.4-1.0.0.4-rhel7.3-ppc64le.iso
```````
